runtime/backend/interface.cpp

```diff
@@ -9,21 +9,86 @@
 #include <executorch/runtime/backend/interface.h>
 #include <executorch/runtime/platform/assert.h>
 
-namespace torch {
-namespace executor {
-
-PyTorchBackendInterface::~PyTorchBackendInterface() {}
+#ifdef _WIN32
+#include <memory>
+#include <windows.h>
+#include <tchar.h>
+#define getpid GetCurrentProcessId
+#else
+#include <unistd.h>
+#endif
 
 // Task t128866626: Remove global static variables.
 // We want to be able to run multiple Executor instances
 // and having a global registration isn't a viable solution
 // in the long term.
-BackendRegistry& getBackendRegistry();
-BackendRegistry& getBackendRegistry() {
-  static BackendRegistry backend_reg;
+#ifdef _WIN32
+
+#define SHARED_MEMORY_NAME "torch_executor_backend_registry"
+static std::shared_ptr<torch::executor::BackendRegistry> backend_reg;
```

Comment:
cc @cccclai  on the backend changes. Personally seeing shared_ptr in the core runtime is scary to me even if its gated behind windows which afaik should always have malloc and atomic available